File: android/src/main/java/com/opentokreactnative/OpentokReactNativePackage.java
package com.opentokreactnative;

import com.facebook.react.BaseReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.model.ReactModuleInfo;
import com.facebook.react.module.model.ReactModuleInfoProvider;

import java.util.HashMap;
import java.util.Map;

public class OpentokReactNativePackage extends BaseReactPackage {
    public NativeModule getModule(String name, ReactApplicationContext reactContext) {
        if (name.equals(OpentokReactNativeModule.NAME)) {
            return new OpentokReactNativeModule(reactContext);
        } else {
            return null;
        }
    }

    @Override
    public ReactModuleInfoProvider getReactModuleInfoProvider() {
        return new ReactModuleInfoProvider() {
            @Override
            public Map<String, ReactModuleInfo> getReactModuleInfos() {
                Map<String, ReactModuleInfo> map = new HashMap<>();
                map.put(OpentokReactNativeModule.NAME, new ReactModuleInfo(
                        OpentokReactNativeModule.NAME, // name
                        OpentokReactNativeModule.NAME, // className
                        false,  // canOverrideExistingModule
                        false,  // needsEagerInit
                        false,  // isCxxModule
                        true // isTurboModule
                ));
                return map;
            }
        };
    }
}

File: android/src/main/java/com/opentokreactnative/OpentokReactNativePackage.kt
package com.opentokreactnative

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider
import java.util.HashMap

class OpentokReactNativePackage : BaseReactPackage() {
  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == OpentokReactNativeModule.NAME) {
      OpentokReactNativeModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      val moduleInfos: MutableMap<String, ReactModuleInfo> = HashMap()
      moduleInfos[OpentokReactNativeModule.NAME] = ReactModuleInfo(
        OpentokReactNativeModule.NAME,
        OpentokReactNativeModule.NAME,
        false,  // canOverrideExistingModule
        false,  // needsEagerInit
        true,  // hasConstants
        false,  // isCxxModule
        true // isTurboModule
      )
      moduleInfos
    }
  }
}
